### ISSUE TITLE
Implement Go 1.10 changes

### ISSUE THREAD

**Issue by johanbrandhorst** [jackc/pgx#380]
Go 1.10 comes with some exciting changes to `database/sql/driver`:

https://tip.golang.org/doc/go1.10#database/sql/driver

To mention a few that would be interesting:

> Drivers that want to construct a sql.DB for their clients can now implement the Connector interface and call the new sql.OpenDB function, instead of needing to encode all configuration into a string passed to sql.Open.

>  Drivers that implement ExecerContext no longer need to implement Execer; similarly, drivers that implement QueryerContext no longer need to implement Queryer. Previously, even if the context-based interfaces were implemented they were ignored unless the non-context-based interfaces were also implemented. 

These things could probably be further split into separate issues for implementation.

**Comment by james-lawrence** [comment]
I'll likely tackle the OpenDB stuff as the reason I proposed that change was for pgx. but going to wait until 1.10 is actually released.

**Comment by qJkee** [comment]
Go 1.10 Released!

**Comment by johanbrandhorst** [comment]
@james-lawrence any progress on this?

**Comment by james-lawrence** [comment]
PR is open, its on @jackc atm to merge or let me know what to change.

**Comment by jackc** [comment]
Sorry for the delay. PR is merged now.

**Comment by mohanraj-r** [comment]
I am not able to use the `stdlib.OpenDB()` on Go1.10
Not sure what I am not missing .. 

```
$ go version
go version go1.10.3 darwin/amd64

# Make sure the OpenDB function is present in the package on my system
$ godoc github.com/jackc/pgx/stdlib | grep OpenDB
func OpenDB(config pgx.ConnConfig, opts ...OptionOpenDB) *sql.DB
type OptionOpenDB func(*connector)
    OptionOpenDB options for configuring the driver when opening a new db
func OptionAfterConnect(ac func(*pgx.Conn) error) OptionOpenDB

# Make sure the opendb file is included in build
$ go list -f '{{.GoFiles}}' github.com/jackc/pgx/stdlib
[opendb.go sql.go]

# run my test
$ go test -v  ./...
...
./sql.go:54:7: undefined: stdlib.OpenDB
...
```

sql.go:
```go
	connConfig := pgx.ConnConfig{
		...
	}

         db = stdlib.OpenDB(connConfig)
```

**Comment by james-lawrence** [comment]
odd I'll do a verification this weekend. but it should be working, I'd look at your go environment for issues?

**Comment by mohanraj-r** [comment]
Found the issue: a older version of pgx in `vendor/` was overriding the system package 🤦‍♂️

**Comment by jackc** [comment]
Is there anything still remaining for Go 1.10 changes or should this issue be closed?

**Comment by qJkee** [comment]
I think it has to be closed because go 1.11 will be released this month, and with 1.10 and it's features it's seems ok

**Comment by james-lawrence** [comment]
as far as the sql.OpenDB is concerned yes. I assume nothing needs to do anything about the Execer/Queryer unless we want to.
